### PR TITLE
Small doc to track common build and installation errors for iOS apps.

### PR DESCRIPTION
**Pull request by ButkiewiczP**
Adds a small document to track common build errors, and common installation errors for iOS apps. Contains a short list right now, but we'll fill it in as we encounter more errors.

### CODE REVIEW

**Review by alexpersian — APPROVED** [review]

File: ios/build_and_install_errors.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+##### Q. My tests don't run and I see the error "iOSSimulator: Could not launch simulator: -10810"
+A. Error "-10810" means that the SystemUIServer process is not running for the current user. You'll see this error when trying to run unit tests as a user who is **not** currently logged into the desktop (i.e. doesn't have a UI server). On Jenkins, this means that the Jenkins user executing the iOS simulator must be set to auto-login. A node reboot is usually required for nodes experiencing this issue.
+
+## Installation Errors
+
+There are some common pitfalls when installing applications from the OTA server. Here are some things to be aware of if you run into problems:
+
+- An OTA application cannot be installed over an exisiting App Store version of the same application.
+- Ensure the phone your installing the app on meets the hardware requirements specified by the app itself. For example, if the app requires GPS to work, you won't be able to install it on older iPod touch devices or devices without any GPS.
```
Comment (nivektric) [inline ios/build_and_install_errors.md]: your --> you're

File: ios/build_and_install_errors.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+##### Q. My tests don't run and I see the error "iOSSimulator: Could not launch simulator: -10810"
+A. Error "-10810" means that the SystemUIServer process is not running for the current user. You'll see this error when trying to run unit tests as a user who is **not** currently logged into the desktop (i.e. doesn't have a UI server). On Jenkins, this means that the Jenkins user executing the iOS simulator must be set to auto-login. A node reboot is usually required for nodes experiencing this issue.
+
+## Installation Errors
+
+There are some common pitfalls when installing applications from the OTA server. Here are some things to be aware of if you run into problems:
+
+- An OTA application cannot be installed over an exisiting App Store version of the same application.
```
Comment (bobgilmore) [inline ios/build_and_install_errors.md]: Can you describe the error mode a bit more here? What actually happens or doesn't happen? (do you get the confirmation dialog, but then still have the old version of the app, or do you not even *get* the confirmation dialog, or what?)

File: ios/build_and_install_errors.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+When building your project locally, or remotely on a CI server, there are some common errors you may run in to. Similarly, installing ad-hoc and distribution apps from an OTA server can also pose some problems. Some errors can be fixed within your project configuration, or within the code itself -- however, some errors may require administrator privileges.
+
+## Build Errors
+
+##### Q. My build fails with "ssh: Could not resolve hostname"
+A. This is a general network error and indicates the network dropped at some point in the build. Restart the build to try it again.
+
+##### Q. I see a lot of "error: unable to read module map contents"
+A. This error happens when there was an issue installing Cocoapod dependencies. Ensure there weren't any odd changes to your Podfile committed. If you're referencing code that's been pushed since midnight of the last day, it's possible that the Cocoapods master-repo hasn't been updated yet, and doesn't contain spec information for the pod you're referencing.
```
Comment (brightredchilli) [inline ios/build_and_install_errors.md:11]: is this really possible though? in theory if this happens shouldnt we be getting cocoapods errors like 'could not resolve dependency..'?

File: ios/build_and_install_errors.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+##### Q. My tests don't run and I see the error "iOSSimulator: Could not launch simulator: -10810"
+A. Error "-10810" means that the SystemUIServer process is not running for the current user. You'll see this error when trying to run unit tests as a user who is **not** currently logged into the desktop (i.e. doesn't have a UI server). On Jenkins, this means that the Jenkins user executing the iOS simulator must be set to auto-login. A node reboot is usually required for nodes experiencing this issue.
+
+## Installation Errors
+
+There are some common pitfalls when installing applications from the OTA server. Here are some things to be aware of if you run into problems:
+
+##### Q. I receive an error message immediately after tapping "Install" on the OTA
+A. There's a number of factors that can contribute to this.
+- An OTA application cannot be installed over an exisiting App Store version of the same application.
```
Comment (brightredchilli) [inline ios/build_and_install_errors.md]: money

File: ios/build_and_install_errors.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+- Ensure the phone you're installing the app on meets the hardware requirements specified by the app itself. For example, if the app requires GPS to work, you won't be able to install it on older iPod touch devices or devices without any GPS.
+- Ensure the phone meets the minimum OS Version requirements for the app.
+- If you still have issues, read the [OTA Server](https://github.com/IntrepidPursuits/shhherpa/Infrastructure/OTA) Readme for information on what might be going wrong server side.
+
+
+##### Q. My application begins loading on the phone, but fails around 75%
+A. This is generally a codesigning issue. Specifically, this is common when the provisioning profile bundled with the app doesn't match the bundle identifier.
```
Comment (brightredchilli) [inline ios/build_and_install_errors.md:43]: maybe worth to add one more codesigninng related problem - 'my application downloads, but when i start it, it immediately quits'. not to be confused with a crash in the app delegate, of course, but improper codesigning can cause it.

**Review by brightredchilli — APPROVED** [review]
one important comments regarding cocoapods..

File: ios/build_and_install_errors.md (hunk 4 above)
Comment (brightredchilli) [reply ios/build_and_install_errors.md]: oh one more spelling issue exisiting -> existing

**Comment by alexpersian** [comment]
Another issue that's not covered here is if the user attempts an OTA download and seemingly nothing happens it's most likely an issue with 32/64bit builds. This would only appear on older devices that are still 32bit. There's no way to know what's happening outside of pulling the logs directly from device, so this can be frustrating without a prior heads up.

Solution is to just make sure Xcode is building both 32 and 64 bit binaries when uploading to the OTA.

File: ios/build_and_install_errors.md (hunk 3 above)
Comment (ButkiewiczP) [reply ios/build_and_install_errors.md:11]: We do get a failed error, but it's not very easy to find in the logs. These errors show up in bold, red text everywhere in the log, so people more often ask about them.

**Comment by ButkiewiczP** [comment]
Anymore comments here? Would like to get this merged in soon.

**Pull request merged**